src/lib.rs

```diff
@@ -352,6 +352,10 @@ pub enum SynthesisError {
     MalformedSrs,
     #[error("non power of two proofs given for aggregation")]
     NonPowerOfTwo,
+    #[error("incompatible vector length")]
+    InvalidLengthVector,
+    #[error("invalid pairing")]
+    InvalidPairing,
```

Comment:
is the pairing really invalid or is it a pairing check that failed?

Reply:
For some reason the `E::final_exponentiation` calls returns an option (I dont know why to be honest, this should always work out, maybe it does if the input is not the output of the miller_loop function call ). But this isn't related to the pairing check, this is thrown only when the pairing computation has failed.